src/sentry/api/endpoints/project_trace_item_details.py

```diff
@@ -15,15 +17,35 @@
 from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.exceptions import BadRequest
 from sentry.models.project import Project
+from sentry.search.eap.types import SupportedTraceItemType
+from sentry.search.eap.utils import translate_internal_to_public_alias
 from sentry.snuba.referrer import Referrer
 from sentry.utils import snuba_rpc
 
 
-def convert_rpc_attribute_to_json(source: dict) -> dict:
-    for k, v in source.items():
-        if k.startswith("val"):
-            return {"type": k[3:].lower(), "value": v}
-    raise BadRequest(f"unknown field in protobuf: {source}")
+def convert_rpc_attribute_to_json(
+    attributes: list[dict],
+) -> Generator[dict]:
+    for attribute in attributes:
+        source = attribute["value"]
+        for k, v in source.items():
+            if k.startswith("val"):
+                val_type = k[3:].lower()
+                column_type: Literal["string", "number"] = "string"
+                if val_type == "str" or val_type == "bool":
+                    column_type = "string"
+                elif val_type == "int" or val_type == "float" or val_type == "double":
+                    column_type = "number"
+                else:
+                    raise BadRequest(f"unknown column type in protobuf: {val_type}")
+
+                name = (
+                    translate_internal_to_public_alias(
+                        attribute["name"], column_type, SupportedTraceItemType.LOGS
```

Comment:
Is this meant to work with spans too? If so, should this be hard coded as `SupportedTraceItemType.LOGS`?